### ISSUE TITLE
Test alternative structures for time-variation in natural mortality and immigration

### ISSUE THREAD

**Issue by ChloeRN** [ChloeRN/VredfoxIPM#20]
The current model attributes all time variation in the system to immigration, which is a comletely "free" parameter. 
When I attempted to constrain immigration previously (using a normal distribution for annual expected immigrant numbers), it did not work out well since immigration "absorbs" a nearly bimodal signature that is likely due to rodent cycles. 

This may be partially because that variation then had "nowhere else to go". That's why it's worth re-trying constraints on immigration in a model with rodent effects, and also in a model that allows for time-variation in natural mortality, at least of juveniles. 
Regarding the latter, an effect of rodent abundance may also not be unreasonable since first-year survival may well be sensitive to rodent abundance. 

So, there are a few different (combinations of) model structures/constraints to test out: 

- [x] Immigration constrained to be a rate (density-dependent)
- [x] Immigration rate with an effect of rodent abundance
- [x] Natural mortality (of juveniles) with a random year effect
- [x] Natural mortality (of juveniles) with an effect of rodent abundance
- [x] Natural mortality with an effect of reindeer carcasses
- [x] Pregnancy rate with an effect of reindeer carcasses

**Comment by ChloeRN** [comment]
Formulating immigration as a rate led to great improvements in the estimation of immigrant numbers, and also resulted in models that run MUCH faster. Details in #22.

**Comment by ChloeRN** [comment]
Models allowing for time variation in mO converged, but resulted in more uncertain population predictions: 
![image](https://github.com/ChloeRN/VredfoxIPM/assets/38600576/28c16858-3bc9-4e83-9ca6-523a809ce83e)

This largerly as a function of increased uncertainty in yearly natural mortality hazard rates: 
![image](https://github.com/ChloeRN/VredfoxIPM/assets/38600576/6c2e85ca-e17c-4ab6-9be0-44e8665b4089)

Other vital rate parameters were hardly affected by this. The only difference was very slightly lower average harvest mortality: 
![image](https://github.com/ChloeRN/VredfoxIPM/assets/38600576/00967c65-2243-47df-baff-85512f2dbef6)

As for adding a rodent effect on natural mortality: I tested this with the continuous covariate and -- little surprisingly -- this effect could not be estimated as different from 0: 
![image](https://github.com/ChloeRN/VredfoxIPM/assets/38600576/d177fe25-4ef7-4fbd-af9f-7cdcb444b229)

At present, I think we may be best of letting natural mortality be constant across time. 
That means that we will not be able to estimate an LTRE contribution for it, but allowing for time variation on a parameter we already have no data on just to end up with more uncertainty seems to make little sense.

**Comment by ChloeRN** [comment]
An effect of rodent abundance on immigration rate was estimated as substantially negative (i.e. the fewer rodents were around in fall in the larger area, the more immigrants came into Varanger): 

![image](https://github.com/ChloeRN/VredfoxIPM/assets/38600576/a56f1c54-0ac9-41c6-bc61-379a0fc3751b)

Accounting for this effect gave a notable increase in precision in several other vital rates and in the estimates of population-level quantities, too:

![image](https://github.com/ChloeRN/VredfoxIPM/assets/38600576/b9a15bad-2ad7-49dc-97e6-a9542d1c0095)

Notably, the effect size and precision are not affected by whether or not we allow for time variation in mO. 

Seems like including a rodent effect on immigration may indeed make sense.

**Comment by ChloeRN** [comment]
We have two types of data on reindeer: 
Municipality-level accounts of reindeer carcasses and weight/percentage of reindeer in red fox stomachs. 

I tested for single effects on Psi and mO, and also an interaction with rodent abundance on Psi:
![image](https://github.com/ChloeRN/VredfoxIPM/assets/38600576/5fe49e3b-4520-48c3-b4dc-e36bf6664f79)

The posteriors yield little evidence for effects here. Uncertainty is large, and the effects not distinguishable from 0. The exception here may be the single (positive) effect of reindeer carcasses on Psi, but precision is low and this also increases uncertainty in other model estimates, e.g. population size:
![image](https://github.com/ChloeRN/VredfoxIPM/assets/38600576/a84b0753-c4a0-4590-8f8f-478c045ee1b8)

I did my tests with reindeer carcasses reported for Varanger and percentage fox stomachs containing reindeer. 
Since the other "flavours" of covariates are highly correlated, it's likely that the exact choice of covariate makes little difference to the estimate. 

Based on this, and in the interest of finalizing the model, my suggestion would be to not pursue reindeer covariates any further for the time being.
